### PR TITLE
Processing submissions to the registry

### PR DESCRIPTION
**Pull request by quartzjer**
Optionally, only verified signed packages can be allowed now.  Also, publishing packages begins the processing process :)

### CODE REVIEW

**Comment by quartzjer** [comment]
this is almost done, one more todo in publishing issues, and one more trigger to make signed packages get re-sync'd automatically

**Comment by mdz** [comment]
It looks like DMBot is still included in the diff

File: Connectors/GitHub/issue.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+*
+* Please see the LICENSE file for more information.
+*
+*/
+
+
+var request = require("request");
+var async = require('async');
+
+exports.sync = function(pi, cb) {
```
Comment (smurthas) [inline Connectors/GitHub/issue.js]: pi is cute and cb is short, but I think both of these make the code significantly less readable to an outsider joining the project. In that we especially want to encourage the community to build Connectors, I would suggest you change these to proccessInfo and callback.

File: Common/node/lcrypto.js [hunk 2]
```diff
@@ -106,3 +111,24 @@ exports.decrypt = function(data) {
     ret += cipher.final("utf8");
     return ret;
 }
+
+exports.sign = function(data) {
+    if (!data) {
+        logger.error("Error signing " + data);
```
Comment (smurthas) [inline Common/node/lcrypto.js:117]: this should really throw an Error

File: Common/node/lcrypto.js [hunk 3]
```diff
@@ -106,3 +111,24 @@ exports.decrypt = function(data) {
...
+        return false;
+    }
+    var signer = crypto.createSign("RSA-SHA256");
+    signer.update(data, "utf8");
+    return signer.sign(idKey, 'base64');
+}
+
+exports.verify = function(data, sig, keys) {
+    if (!data || !sig || !keys || !Array.isArray(keys)) return false;
```
Comment (smurthas) [inline Common/node/lcrypto.js:126]: This also seems like it should throw and Error

File: Common/node/lcrypto.js (hunk 2 above)
Comment (quartzjer) [inline Common/node/lcrypto.js:117]: Disagree, I'm anti-throw-pattern minded, it's a minimal wrapper just mimicking the underlying function.

Eventually the whole crypto stuff could prob mature into the better callback based but it's not ready yet.

File: Connectors/GitHub/issue.js (hunk 1 above)
Comment (quartzjer) [inline Connectors/GitHub/issue.js]: cb is used in other node codebases and is a well known enough shorthand, and also arg has been forming as the object-with-stuff first argument, updated it to that...

**Comment by quartzjer** [comment]
This is ready for some dev review (cc @temas).  To test, just publish one of your apps, which will automatically generate an issue at https://github.com/Singly/apps/issues associated with the publish request (need ui at some point to explain this, #657).

To test the signing, check out https://github.com/Singly/apps and run sign.js path/to/Me/key package@version and enter user:pass for registry to upload the signature (use the package owner or admin for now).

Signature will be uploaded then so that version is verifiable, and if a config.json requireSigned:true is set on locker start, only the signed packages (only a few test ones right now) will show up and be used!

**Comment by temas** [comment]
Giving this a test run.

**Comment by temas** [comment]
Here's what console had on run, no issue was created.

https://gist.github.com/1711732

**Comment by quartzjer** [comment]
That's not an issue with this branch but with publishing over an existing package, look at the err either it was an older ver or published from a diff locker I think.

**Comment by temas** [comment]
Derp, I'm trying to do too many things at once today and moving too fast.  I'll retry in a sec.

**Comment by mdz** [comment]
I tried to get this to work and failed:

31 Jan 11:03:46 - info: registry publishing mdz-linkpic
31 Jan 11:03:46 - info: attempting to publish Me/github/mdz/LinkPic/package.json
31 Jan 11:03:46 - info: adding user ...JSON...
31 Jan 11:03:49 - verbose: upsert request for Me/github/mdz/LinkPic/package.json
31 Jan 11:03:49 - verbose: updating mdz-linkpic
31 Jan 11:03:49 - verbose: received event for service://me/#mdz-linkpic action(update)
31 Jan 11:03:51 - info: Publising the screenshot for mdz-linkpic
31 Jan 11:03:53 - info: Registry done sending screenshot

but nothing happened, no issue was filed. Jer suggested this might be due to the GitHub auth token being read-only (which mine is). Filed as #673

**Comment by temas** [comment]
Was able to run the entire process on a clean locker.

**Pull request merged**